cmd/skopeo/signing.go

```diff
@@ -98,17 +104,45 @@ func (opts *standaloneVerifyOptions) run(args []string, stdout io.Writer) error
 		return fmt.Errorf("Error reading signature from %s: %v", signaturePath, err)
 	}
 
-	mech, err := signature.NewGPGSigningMechanism()
+	var mech signature.SigningMechanism
+	var fingerprints []string
+	if opts.publickeyfile != "" {
+		publicKeys, err := os.ReadFile(opts.publickeyfile)
+		if err != nil {
+			return fmt.Errorf("Error reading public keys from %s: %v", opts.publickeyfile, err)
+		}
+		mech, fingerprints, err = signature.NewEphemeralGPGSigningMechanism(publicKeys)
+	} else {
+		mech, err = signature.NewGPGSigningMechanism()
+	}
 	if err != nil {
 		return fmt.Errorf("Error initializing GPG: %v", err)
 	}
 	defer mech.Close()
+
+	if opts.publickeyfile != "" && expectedFingerprint == "any" {
+		_, expectedFingerprint, err = mech.Verify(unverifiedSignature)
+		if err != nil {
+			return fmt.Errorf("Could not determine fingerprint from signature: %v", err)
+		}
+		found := false
+		for _, fingerprint := range fingerprints {
+			if expectedFingerprint == fingerprint {
+				found = true
+			}
+		}
+		if !found {
+			// This is theoretically impossible because mech.Verify only works if it can identify the key based on the signature
+			return fmt.Errorf("Signature fingerprint not found in public key file: %s", expectedFingerprint)
+		}
+	}
+
 	sig, err := signature.VerifyDockerManifestSignature(unverifiedSignature, unverifiedManifest, expectedDockerReference, mech, expectedFingerprint)
 	if err != nil {
 		return fmt.Errorf("Error verifying signature: %v", err)
 	}
 
-	fmt.Fprintf(stdout, "Signature verified, digest %s\n", sig.DockerManifestDigest)
+	fmt.Fprintf(stdout, "Signature verified using fingerprint %s, digest %s\n", expectedFingerprint, sig.DockerManifestDigest)
```

Comment:
I’m not too sure about this addition… especially I’d like to avoid a policy tool being built on top, parsing this text; human-targeted texts are not a stable API.

Yes, it’s useful for debugging … but for debugging, also available without using `skopeo`.

Reply:
>  because the message format just changed and broke them :)

… like our own tests, yes :)

Fair enough, let’s leave it in.